### ISSUE TITLE
Issues while training a custom dataset

### ISSUE THREAD

**Issue by b-mutawak** [ultralytics/yolov5#587]
## ❔Question
I have a dataset of around 32K training samples and 4K validation. I'm using the provided yolov5m.pt checkpoint and training from there. For some reason, when I get to the testing portion of each epoch my processing speed slows to a crawl (5-10s /it in testing vs. 2.5 it/s in training for a batch size of 16). I don't see these issues when validating using test.py, this only appears when I'm training and get to the testing portion of each epoch. I've tried this using both V1 and V2 of this Yolov5 implementation. Any ideas what could be causing this?

## Additional context
For context, I'm using a 2080Ti with 256GB of DDR4 RAM. This was tested on both Windows 10 and Ubuntu 18.04, both environments being set up using the requirements.txt. Important to note that reducing my number of training samples mitigates this issue and it wasn't as prevalent at 8K training samples. Putting print statements everywhere shows that the test dataloader and sending the inputs through the model are both taking much longer in testing vs. training.

**Comment by glenn-jocher** [comment]
@b-mutawak testing time can easily be affected by your hyperparameters. Sometimes early epochs produce many false positives (thousands per image) which must all undergo NMS. If you are using all default hyps then you can simply omit testing by `python train.py --notest`.

**Comment by b-mutawak** [comment]
@glenn-jocher Got it. Is testing still necessary if I'm not using the default hyper parameter values?  At what point is it required?

**Comment by glenn-jocher** [comment]
@b-mutawak testing is not required for training. 

--notest will test once at the very end of training rather than every epoch.

**Comment by b-mutawak** [comment]
@glenn-jocher Got it, appreciate the help.

**Comment by watertianyi** [comment]
@glenn-jocher training a custom dataset，I have a dataset of around 700 training samples and 190 test data. I'm using the provided yolov5s.pt checkpoint and training from there.
My results are as follows，are the following results normal? Why are classification and val classification blank? Thank you!
![image](https://user-images.githubusercontent.com/29244715/89745592-92001180-dae7-11ea-8932-a6df1fa69287.png)

**Comment by glenn-jocher** [comment]
Looks typical. If you don’t see classifications metrics your dataset requires no classification.

**Comment by watertianyi** [comment]
@glenn-jocher My data is just one category. Is that normal

**Comment by Ownmarc** [comment]
Yes, classification is when you assign a class to a detection, since they are all the same class, the model can never be wrong on class and then the classification metric becomes meaningless

**Comment by MLcodie** [comment]
Hi, 

I know this thread is stale but I am seeing the similar issue as @hande6688 mentioned but for me val Box and val Classification graphs are empty. I have 15 classes for my custom training. Want to make sure this if this is normal?

![image](https://user-images.githubusercontent.com/75185957/101403687-6f5de980-388a-11eb-801b-f0026a451d9e.png)
